### DISCUSSION TITLE
Prevent user from entering decimal point in the NumberField

### DISCUSSION THREAD

**Discussion by jonathandsouza** [adobe/react-spectrum#7341] [Q&A]
Hi There, 

I have a use case where I need to only allow integers in NumberField (react-aria-components) i.e the user should not be allowed to enter decimal point in the component. 

Any help/guidance would be much appreciated.

**Comment by snowystinger** [comment]
Unfortunately, there's not much we can do about this. Depending on the locale, a decimal point may either be a `,` or a `.`, or something else entirely. Any of those characters could also potentially be the group separator. We do not know what the user most likely intends until the number is complete.

**Reply by cheema-corellian** [reply]
Perhaps this behavior could be controlled with a setting for the NumberField Component?

**Comment by jonathandsouza** [comment]
Hi There, 

Thanks for getting back to me. 

I found a workaround to prevent user from entering decimal point. I achieved this by setting the [maximumFractionDigits](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat#maximumfractiondigits) to 0 and [useGrouping](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat#usegrouping) to false. 

The downside is that the user will not be able to do grouping of digits, which is acceptable in my use case. 


https://stackblitz.com/edit/vitejs-vite-hfjf78?file=src%2FApp.tsx
